src/style-spec/function/evaluation_context.js

```diff
@@ -0,0 +1,170 @@
+'use strict';
+
+const parseColor = require('../util/parse_color');
+const interpolate = require('../util/interpolate');
+
+class RuntimeError extends Error {
+    constructor(message) {
+        super(message);
+        this.name = 'ExpressionEvaluationError';
+        this.message = message;
+    }
+
+    toJSON() {
+        return `${this.name}: ${this.message}`;
+    }
+}
+
+class Color {
+    constructor(input) {
+        if (Array.isArray(input)) {
+            this.value = input;
+        } else {
+            this.value = parseColor(input);
+            if (typeof this.value === 'undefined') throw new RuntimeError(`Could not parse color from value '${input}'`);
+        }
+    }
+}
+
+module.exports = () => ({
+    get: function (obj, key) {
+        if (!obj) throw new RuntimeError(`Cannot get property ${key} from null object`);
+        if (!(key in obj)) throw new RuntimeError(`Property ${key} not found in object with keys: [${Object.keys(obj)}]`);
+        return obj[key];
+    },
+
+    typeOf: function (x) {
+        if (Array.isArray(x)) return 'Vector<Value>';
```

Comment:
Or, if we think `T[]` is _too_ concise and easy to misread, we could do:
- Vector<T>: `Array of T's`
- Array<T, N>: `Array of N T's`